node-red-contrib-botbuilder/msbot/actions.js

```diff
@@ -1,32 +1,140 @@
-const { BotFrameworkAdapter, TurnContext } = require("botbuilder");
+const { BotFrameworkAdapter, TurnContext, MemoryStorage, ConversationState, SkillHttpClient, SkillHandler, ChannelServiceRoutes, ActivityTypes } = require("botbuilder");
+const { SimpleCredentialProvider, AuthenticationConfiguration } = require('botframework-connector');
 const { VBMBot } = require("./botVBM");
 const getMessage = require("./messages.js");
-
-
+const { SkillsConfiguration } = require('./skillsConfiguration');
+const { SkillConversationIdFactory } = require('./skillConversationIdFactory');
+const { AllowedCallersClaimsValidator } = require('./allowedCallersClaimsValidator');
 const helper = require("node-red-viseo-helper");
 const botmgr = require("node-red-viseo-bot-manager");
 
-
 let adapter;
-let bot;
+let skillEndpoint;
+let skillHandler;
 
-async function initConnector(config, node, startCmd) {
-  return new Promise( function (resolve, reject) {
+const getAuthConfig = (config, allowedSkills) => {
+  if (config.botType === 'rootBot' || config.botType === 'skillBot') {
+    const allowedCallersClaimsValidator = new AllowedCallersClaimsValidator(allowedSkills);
+
+    return new AuthenticationConfiguration([], (claims) => {
+      allowedCallersClaimsValidator.validateClaims(claims);
+    });
+  }
+};
+
+function createAdapter(config, authConfig) {
+  let adapter;
+  let botbuilderConfig;
 
-    let botbuilderConfig = {
+  if (config.botType === 'none' || config.botType === '') {
```

Comment:
you can simplify this code by creating a botbuilderConfig object containing appId and appPassword in all cases. Then if the type is root or skill, you just add the authConfig parameter to this object.